### PR TITLE
cmd/cleanup: add socketlb program cleanup

### LINKED ISSUE #10067: Remove attached bpf_sock.o when doing cilium cleanup
`bpf/init.sh` (grep for `bpf_load_cgroup`) might attach (depending on a configuration) BPF programs to the root cgroup.

`cilium cleanup` should detect and remove those programs.

### PR DESCRIPTION
**Pull request by rgo3**
This leverages this newly added `socketlb` package API [[1]](https://github.com/cilium/cilium/blob/f41d69c9769ae8e6f4ebea606f0641b07eabfe05/pkg/socketlb/socketlb.go#L165) to remove any leftover socketlb bpf programs.

Fixes: #10067

### CODE REVIEW

**Comment by rgo3** [comment]
Marking as ready for review now that I have manually tested this with a minikube setup and successfully removed socketlb progs from a node by running `cilium cleanup`.

**Review by asauber — APPROVED** [review]
LGTM

File: cilium/cmd/cleanup.go [hunk 1]
```diff
@@ -249,10 +251,14 @@ func (c ciliumCleanup) cleanupFuncs() []cleanupFunc {
 	cleanupTCFilters := func() error {
 		return removeTCFilters(c.tcFilters)
 	}
+	cleanupSocketLBPrograms := func() error {
```
Comment (lmb) [inline cilium/cmd/cleanup.go]: Nit: removeSocketLBPrograms is already `func() error` so the closure isn't really necessary?

**Review by lmb — APPROVED** [review]

**Comment by rgo3** [comment]
ci-mulitcluster failed with #25064, rerunning.

**Pull request merged**